app/src/main/java/de/westnordost/streetcomplete/MainActivity.java

```diff
@@ -247,6 +252,36 @@ public void onServiceDisconnected(ComponentName className)
 		{
 			questController.deleteOld();
 		}
+
+		Intent intent = getIntent();
+		Uri data = intent.getData();
+		if (Intent.ACTION_VIEW.equals(intent.getAction()) && data != null)
+		{
+			if ("geo".equals(data.getScheme()))
+			{
+				double latitude = 0;
+				double longitude = 0;
+				float zoom = -1;
+
+				String geoUriRegex = "^geo:(-?[0-9]*\\.?[0-9]+),(-?[0-9]*\\.?[0-9]+).*?(?:\\?z=([0-9]*\\.?[0-9]+))?";
+				Pattern pattern = Pattern.compile(geoUriRegex);
+				Matcher matcher = pattern.matcher(data.toString());
+				while (matcher.find())
+				{
+					latitude = Double.parseDouble(matcher.group(1));
+					longitude = Double.parseDouble(matcher.group(2));
+					if (matcher.group(3) != null) {
+						zoom = Float.valueOf(matcher.group(3));
+					}
```

Comment:
A faulty URL should be handled gracefully here. I'd wrap the parsing into a try-catch

Reply:
Mmhh... But with a wrong URL, the regex match will fail too, so if [`matcher.matches()`](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Matcher.html#matches--) returns false, nothing else is done...

Reply:
Right, I thought maybe it would be possible to smuggle something like `0.1.234` in there, but the regex looks sound.